File: p2341/src/main.js
function numberOfPairs(nums) {
    let result = [0, 0];

    let count = {};

    for (let i = 0; i < nums.length; i++) {
        if (count[nums[i]]) count[nums[i]]++
        else count[nums[i]] = 1
    }

    for (const [key, value] of Object.entries(count)) {
        result[0] += Math.floor(value / 2)
        result[1] += value % 2
    }

    return result
};

console.log(numberOfPairs([1,3,2,1,3,2,2]))

File: p2341/src/main.rs
use std::collections::HashMap;

struct Solution {}

impl Solution {
    pub fn number_of_pairs(nums: Vec<i32>) -> Vec<i32> {
        let mut result = vec![0, 0];

        let mut count = HashMap::new();

        for i in nums {
            let mut stm = count.get(&i);
            match stm {
                Some(v) => count.insert(&stm, Some(v + 1)),
                None => count.insert(&stm, Some(1)),
            };
            
            // count.insert(&i, count.get(&i).unwrap() + 1);
        }

        println!("{:?}", count);

        return result;
    }
}

fn main() {
    println!("{:?}", Solution::number_of_pairs(vec![1,3,2,1,3,2,2]));
}
